doc/api/vm.md

```diff
@@ -1555,13 +1591,66 @@ According to the [V8 Embedder's Guide][]:
 > JavaScript applications to run in a single instance of V8. You must explicitly
 > specify the context in which you want any JavaScript code to be run.
 
-When the method `vm.createContext()` is called, the `contextObject` argument
-(or a newly-created object if `contextObject` is `undefined`) is associated
-internally with a new instance of a V8 Context. This V8 Context provides the
-`code` run using the `node:vm` module's methods with an isolated global
-environment within which it can operate. The process of creating the V8 Context
-and associating it with the `contextObject` is what this document refers to as
-"contextifying" the object.
+When the method `vm.createContext()` is called with an object, the `contextObject` argument
+will be used to wrap the global object of a new instance of a V8 Context
+(if `contextObject` is `undefined`, a new object will be created from the current context
+before its contextified). This V8 Context provides the `code` run using the `node:vm`
+module's methods with an isolated global environment within which it can operate.
+The process of creating the V8 Context and associating it with the `contextObject`
+is what this document refers to as "contextifying" the object.
+
+The contextifying would make the global object within the V8 context behave differently
+from a normal global object. For example, it is not reference equal to the `globalThis`
+value in the context, and it cannot be freezed.
+
+```js
+const vm = require('node:vm');
+
+// An undefined `contextObject` option makes the global object contextified.
+let context = vm.createContext();
+console.log(vm.runInContext('globalThis', context) === context);  // false
+// A contextified global object cannot be freezed.
+try {
+  vm.runInContext('Object.freeze(globalThis);', context);
+} catch(e) {
+  console.log(e); // TypeError: Cannot freeze
+}
+console.log(vm.runInContext('globalThis.foo = 1; foo;', context));  // 1
+```
+
+To create a vanilla context with an ordinary global object, specify `vm.constants.VANILLA_CONTEXT`
+as the `contextObject` argument.
+
+### `vm.constants.VANILLA_CONTEXT`
```

Comment:
Is "vanilla" a standard term for this concept? I'm wondering if it's worth being more verbose here, e.g. something like `CONTEXT_WITHOUT_INTERCEPTORS`, or generally something that makes it possible to infer properties of what's happening in code using it without having to consult the Node.js docs

Reply:
Maybe `USE_ORIGINAL_GLOBAL_OBJECT` or `NO_CONTEXTIFYING`? We have never mentioned the interceptors in the docs so far, on the other hand describing how it works in the docs now would be like describing the implementation details that most users would not care about. The term that the documentation has been using to describe what happens usually is that the wrapper object is "contextified" (not really an intuitive term that you can understand without looking at the docs, but then this term has been there for so long that I think most vm context users should be aware of it already).

Reply:
Sure, these work for me, I'm just looking for potential other options 🙂 I guess the broader word for what we call "contextifying" is just "wrapping" but that might just be a bit too generic to be helpful here.

Reply:
Yeah I was a bit skeptical about "vanilla" too, updated to `DONT_CONTEXTIFY` which I think is probably the easiest to understand out of the options so far.